staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -180,6 +192,28 @@ type SharedInformer interface {
 	// The handler should return quickly - any expensive processing should be
 	// offloaded.
 	SetWatchErrorHandler(handler WatchErrorHandler) error
+
+	// EventHandlerCount return the number of actually registered
+	// event handlers.
+	EventHandlerCount() int
+
+	// IsStopped reports whether the informer has already been stopped.
+	// Adding event handlers to already stopped informers is not possible.
+	IsStopped() bool
+
+	// IsStarted reports whether the informer has already been started
+	IsStarted() bool
```

Comment:
The problem is, that they should be used outside the package, in a higher informer management layer, therefore they cannot be private. They will not only be used for testing (see my comment above)

Reply:
We need to refactor the interface until the things people need to do can be done race-free. E.g. instead of calling IsStopped() before adding a handler, just add the handler and it gives you an error if the informer has been stopped.